Assets/MixedRealityToolkit.SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2232 @@
+                    //use a dummy bounds of one to get the local scale to match;
+                    clippingBounds.size = new Vector3((clippingBounds.size.x * Tiers), (clippingBounds.size.y * ViewableArea), clippingBounds.size.z);
+                    clipBox.transform.localScale = new Bounds(Vector3.zero, Vector3.one).ScaleFromBounds(clippingBounds, OcclusionScalePadding);
+
+                    //adjust where the center of the clipping box is
+                    viewableCenter.x = (clipBox.transform.localScale.x * 0.5f) - (OcclusionScalePadding.x * 0.5f) + OcclusionPositionPadding.x;
+                    viewableCenter.y = (((clipBox.transform.localScale.y * 0.5f) - (OcclusionScalePadding.y * 0.5f)) * -1) + OcclusionPositionPadding.y;
+                    viewableCenter.z = OcclusionPositionPadding.z;
+                    break;
+
+                case ScrollDirectionType.LeftAndRight:
+
+                    //Same as above for L <-> R
+                    clippingBounds.size = new Vector3(clippingBounds.size.x * ViewableArea, clippingBounds.size.y * Tiers, clippingBounds.size.z);
+                    clipBox.transform.localScale = new Bounds(Vector3.zero, Vector3.one).ScaleFromBounds(clippingBounds, OcclusionScalePadding);
+
+                    //Same as above for L <-> R
+                    viewableCenter.x = (clipBox.transform.localScale.x * 0.5f) - (OcclusionScalePadding.x * 0.5f) + OcclusionPositionPadding.x;
+                    viewableCenter.y = ((clipBox.transform.localScale.y * 0.5f) - (OcclusionScalePadding.y * 0.5f) + OcclusionPositionPadding.y) * -1.0f;
+                    viewableCenter.z = OcclusionPositionPadding.z;
+                    break;
+            }
+
+            //Apply our new values
+            clipBox.transform.localPosition = viewableCenter;
+
+            //add our objects to the clippingBox queue
+            AddAllItemsToClippingObject();
+        }
+
+        #endregion ObjectCollection methods
+
+        #region Monobehavior Implementation
+
+        private void OnEnable()
+        {
+            //Register for global input events
+            if (CoreServices.InputSystem != null)
+            {
+                CoreServices.InputSystem.RegisterHandler<IMixedRealityInputHandler>(this);
+                CoreServices.InputSystem.RegisterHandler<IMixedRealityTouchHandler>(this);
+                CoreServices.InputSystem.RegisterHandler<IMixedRealityPointerHandler>(this);
+                CoreServices.InputSystem.RegisterHandler<IMixedRealitySourceStateHandler>(this);
+            }
+
+            if (useOnPreRender)
+            {
+                if (clippingObject == null || clipBox == null)
+                {
+                    Debug.Log(name + " UseOnPreRender is enabled, but ScrollingObjectCollection needs UpdateCollection() to be called first.");
+                    return;
+                }
+
+                clipBox.UseOnPreRender = true;
```

Comment:
Should this also be in the UseOnPreRender setter?

Reply:
good catch.